### PR TITLE
Better dedupe for search results

### PR DESCRIPTION
**Pull request by htanjo**
### Issue
Since [`dedupe()`](https://github.com/bower/search/pull/47) was implemented, some packages are completely missing and the search result shows unofficial name packages. For example:
- **`select2`** - listed as `select2-ng`
- **`polymer`** - listed as `polymer-polymer`
- **`bootstrap-sass`** - listed as `twitter-bootstrap-sass`
- **`colors`** - listed as `mrmrs-colors`

I know we can fix them using `whitelist.js` one by one, but it's troublesome.

### Solution
This patch prioritizes package of **same name with its repository**, if duplicated.
Mostly it should return better results.

### CODE REVIEW

File: js/controllers/index_controller.js [hunk 1]
```diff
@@ -126,9 +126,24 @@ module.exports = function (App) {
...
-				return _.uniq(matchedResults.reverse(), function(item){
+				var groupedResults = _.groupBy(matchedResults.reverse(), function (item) {
 					return item.website;
 				});
+				var list = [];
+				_.forEach(groupedResults, function (group) {
+					var matchedItem;
+					if (group.length > 1) {
+						matchedItem = _.find(group, function (item) {
+							var repoName = item.website.split('/').pop();
```
Comment (davidjbradshaw) [inline js/controllers/index_controller.js]: Can this line be moved outside the function, as aren't all items in the group going to have the same URL?

    repoName = group[0].website.split('/').pop();

**Comment by htanjo** [comment]
Exactly! It's faster. Thanks @davidjbradshaw.

**Comment by davidjbradshaw** [comment]
Thanks for this.

**Comment by htanjo** [comment]
No problem! Thanks for merging.

**Pull request merged**